tst/EnergyPlus/unit/WaterThermalTanks.unit.cc

```diff
@@ -2053,3 +2053,232 @@ TEST_F(EnergyPlusFixture, StratifiedTankCalc)
+        "    5.317076448,             !- Cooling Power Consumption Coefficient 3",
+        "    0,                       !- Cooling Power Consumption Coefficient 4",
+        "    0,                       !- Cooling Power Consumption Coefficient 5",
+        "    1000,                    !- Nominal Time for Condensate Removal to Begin {s}",
+        "    1.5;                     !- Ratio of Initial Moisture Evaporation Rate and Steady State Latent Capacity {dimensionless}",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool ErrorsFound = false;
+    HeatBalanceManager::GetZoneData(ErrorsFound); // read zone data
+    EXPECT_FALSE(ErrorsFound);
+
+    DataGlobals::NumOfTimeStepInHour = 1; // must initialize this to get schedules initialized
+    DataGlobals::MinutesPerTimeStep = 60; // must initialize this to get schedules initialized
+    ScheduleManager::ProcessScheduleInput();
+    ScheduleManager::ScheduleInputProcessed = true;
+
+    DataGlobals::TimeStep = 1;
+    DataGlobals::HourOfDay = 1;
+    DataEnvironment::Month = 7;
+    DataEnvironment::DayOfMonth = 21;
+    DataGlobals::HourOfDay = 1;
+    DataEnvironment::DSTIndicator = 0;
+    DataEnvironment::DayOfWeek = 2;
+    DataEnvironment::HolidayIndex = 0;
+    DataEnvironment::DayOfYear_Schedule = General::OrdinalDay(DataEnvironment::Month, DataEnvironment::DayOfMonth, 1);
+    ScheduleManager::UpdateScheduleValues();
+
+    int TankNum(1);
+    ErrorsFound = false;
+    EXPECT_FALSE(WaterThermalTanks::GetWaterThermalTankInputData(ErrorsFound));
+    EXPECT_EQ(DataHeatBalance::HeatReclaimSimple_WAHPCoil(1).Name, "GSHP_COIL1");
+    EXPECT_EQ(DataHeatBalance::HeatReclaimSimple_WAHPCoil(1).SourceType, "Coil:Cooling:WaterToAirHeatPump:EquationFit");
+
+    WaterThermalTanks::WaterThermalTankData &Tank = WaterThermalTank(TankNum);
+    WaterThermalTanks::WaterHeaterDesuperheaterData &Desuperheater = WaterHeaterDesuperheater(Tank.DesuperheaterNum);
+
+    for (int i = 1; i <= Tank.Nodes; ++i) {
+        auto &node = Tank.Node(i);
+        node.SavedTemp = 39;
+        node.Temp = 39;
+    }
+
+    Tank.TankTemp = 39;
+    Tank.AmbientTemp = 20.0;
+    Tank.UseInletTemp = 10;
+    Tank.UseMassFlowRate = 0.003;
+    Tank.SourceOutletTemp = 39;
+    Tank.SourceMassFlowRate = 0.003;
+    Tank.TimeElapsed = 0.0;
+    Desuperheater.SetPointTemp = 60;
+    Desuperheater.Mode = 1;
+    Node(Desuperheater.WaterInletNode).Temp = Tank.SourceOutletTemp;
+    
+    HourOfDay = 0;
+    TimeStep = 1;
+    TimeStepZone = 1. / 60.;
+    TimeStepSys = TimeStepZone;
+    SysTimeElapsed = 0.0;
+    DataHeatBalance::HeatReclaimSimple_WAHPCoil(1).AvailCapacity = 1000;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(1).PartLoadRatio = 0.0;
+    WaterThermalTanks::InitWaterThermalTank(TankNum,true);
+    Tank.SetPointTemp = 45;
+    Tank.SetPointTemp2 = 45;
+    WaterThermalTanks::CalcDesuperheaterWaterHeater(TankNum,true);
+
+    // check source inlet and outlet temperatures are different
```

Comment:
Did you mean to check the temperatures here, or is this an old comment?

Reply:
Haha it's an old comment